src/wfs/src/main/java/org/geoserver/wfs/response/ShapeZipOutputFormat.java

```diff
@@ -83,7 +83,7 @@
  * @author ported to gs 1.6.x by Saul Farber, MassGIS, saul.farber@state.ma.us
  */
 public class ShapeZipOutputFormat extends WFSGetFeatureOutputFormat
-        implements ApplicationContextAware {
+        implements ApplicationContextAware, ComplexFeatureAwareFormat {
```

Comment:
The format is not really "aware" on its own here... it's more like "ComplexFeatureUnware" really, and needs other code to help it with conversion from simple to complex.

Maybe you should be marking GML3OutputFormat and GeoJSONOutputFormat as ComplexFeatureAware instead, and the XSLT and OGR output formats, with a default implementation of supportsComplexFeatures that returns true all the time.